src/promptflow/promptflow/_telemetry/activity.py

```diff
@@ -52,19 +55,31 @@ def log_activity(
     :type custom_dimensions: dict
     :return: None
     """
-    custom_dimensions = custom_dimensions or {}
+    from promptflow._core.operation_context import OperationContext
+
+    if not custom_dimensions:
+        custom_dimensions = {}
+
     context = OperationContext.get_instance()
-    request_id = custom_dimensions.get('request_id', context.get('request_id', str(uuid.uuid4())))
     user_agent = context.get_user_agent()
+    # TODO(2699383): use same request id with service caller
+    request_id = request_id_context.get()
+    if not request_id:
+        # public function call
+        first_call = True
+        request_id = str(uuid.uuid4())
+        request_id_context.set(request_id)
+    else:
+        first_call = False
+
     activity_info = {
-        # TODO(2699383): use same request id with service caller
         "request_id": request_id,
+        "first_call": first_call,
         "activity_name": activity_name,
         "activity_type": activity_type,
-        "user_agent": user_agent
+        "user_agent": user_agent,
```

Comment:
Curious about a question, is the last "," automatically added by your IDE or manually added? 
Is this recommended for our python style?

Reply:
Not sure if it's added by pre-commit. Did you setup your local dev env with pre-commit?